api/app/wildfire_one/schema_parsers.py

```diff
@@ -130,20 +131,50 @@ def generate_station_daily(raw_daily, station: WFWXWeatherStation, fuel_type: st
     pc = FUEL_TYPE_LOOKUP[fuel_type]["PC"]
     pdf = FUEL_TYPE_LOOKUP[fuel_type]["PDF"]
     cbh = FUEL_TYPE_LOOKUP[fuel_type]["CBH"]
+    cfl = FUEL_TYPE_LOOKUP[fuel_type]["CFL"]
 
     isi = raw_daily.get('initialSpreadIndex', None)
     bui = raw_daily.get('buildUpIndex', None)
     ffmc = raw_daily.get('fineFuelMoistureCode', None)
     fmc = cffdrs.foliar_moisture_content(station.lat, station.long, station.elevation, get_julian_date_now())
-    sfc = cffdrs.surface_fuel_consumption(fuel_type, bui, ffmc, pc)
+
+    sfc = None
+    try:
+        sfc = cffdrs.surface_fuel_consumption(fuel_type, bui, ffmc, pc)
+    except cffdrs.CFFDRSException as exception:
+        logger.error(exception, exc_info=True)
 
     cc = raw_daily.get('grasslandCuring')
     if cc is None:
         cc = FUEL_TYPE_LOOKUP[fuel_type]["CC"]
-    ros = cffdrs.rate_of_spread(fuel_type, isi, bui, fmc, sfc, pc=pc,
-                                cc=cc,
-                                pdf=pdf,
-                                cbh=cbh)
+
+    ros = None
+    try:
+        if sfc is not None:
+            ros = cffdrs.rate_of_spread(fuel_type, isi, bui, fmc, sfc, pc=pc,
+                                        cc=cc,
+                                        pdf=pdf,
+                                        cbh=cbh)
+    except cffdrs.CFFDRSException as exception:
+        logger.error(exception, exc_info=True)
+
+    cfb = None
+    try:
+        if sfc is not None:
+            cfb = calculate_cfb(fuel_type, fmc, sfc, ros, cbh)
+    except cffdrs.CFFDRSException as exception:
+        logger.error(exception, exc_info=True)
+
+    hfi = None
+    try:
+        if sfc is not None and ros is not None and cfb is not None:
+            hfi = cffdrs.head_fire_intensity(fuel_type=fuel_type,
+                                             percentage_conifer=pc,
+                                             percentage_dead_balsam_fir=pdf,
+                                             ros=ros, cfb=cfb, cfl=cfl, sfc=sfc)
+    except cffdrs.CFFDRSException as exception:
+        logger.error(exception, exc_info=True)
+
     return StationDaily(
         code=station.code,
         status="Observed" if raw_daily.get('recordType', '').get('id') == 'ACTUAL' else "Forecasted",
```

Comment:
for discussion - FireBat just displays "ACTUAL", "FORECAST", or whatever else WFWX happens to give to us, because that's what SS requested. However, modifying "ACTUAL" -> "Observed" and "FORECAST" -> "Forecasted" matches current MoreCast behaviour. I'd imagine that there's some UX advice on this subject regarding consistent behaviour across apps.

Reply:
if a weather station experiences a malfunction and reports out really wacky weather values (or no values), staff can go into WFWX and manually overwrite the dodgy values. In these situations, I thiiiink the status actually shows as "MANUAL" in WFWX, which would get displayed as "Forecasted" with our current logic

Reply:
This is a good one, it's been on my mind. Ideally we are consistent across BCWS tools:
- Forecasters see the same terminology in WildfireOne interface and MoreCast
- Fire Behaviour Analysts see the same terminology across PSU Tools (FireBAT, HFI Calc., etc.)
- - presently, FBANs would use 'forecast' in their HFI Workbook, as they copy the data from WildfireOne 

Follow-up: Not sure why we adjusted this in MoreCast - I'll ask forecasters if actual vs. observed is significant to them/why. 

Recommendation:
I'm inclined to say we stick to "ACTUAL" and "FORECAST" for nowww. 
We should pull that "MANUAL" status across if we can - nice catch.

Reply:
what I propose is that we simply use whatever status we get from WFWX, unmodified